packages/data-context/src/actions/WizardActions.ts

```diff
@@ -63,6 +79,105 @@ export class WizardActions {
+      this.data.detectedLanguage = null
+
+      await this.detectLanguage()
+      debug('detectedLanguage %s', this.data.detectedLanguage)
+      this.data.chosenLanguage = this.data.detectedLanguage || 'js'
+
+      let hasPackageJson = true
+
+      try {
+        await this.ctx.fs.access(path.join(this.ctx.currentProject, 'package.json'), this.ctx.fs.constants.R_OK)
+      } catch (e) {
+        debug('Could not read or find package.json: %O', e)
+        hasPackageJson = false
+      }
+      const packageJson: {
+        dependencies?: { [key: string]: string }
+        devDependencies?: { [key: string]: string }
+      } = hasPackageJson ? await this.ctx.fs.readJson(path.join(this.ctx.currentProject, 'package.json')) : {}
+
+      debug('packageJson %O', packageJson)
+      const dependencies = [
+        ...Object.keys(packageJson.dependencies || {}),
+        ...Object.keys(packageJson.devDependencies || {}),
+      ]
+
+      this.detectFramework(dependencies)
+      debug('detectedFramework %s', this.data.detectedFramework)
+      this.detectBundler(dependencies)
+      debug('detectedBundler %s', this.data.detectedBundler)
+
+      this.data.chosenFramework = this.data.detectedFramework || null
+      this.data.chosenBundler = this.data.detectedBundler || null
+    }
+  }
+
+  private detectFramework (dependencies: string[]) {
+    // Detect full featured frameworks
+    if (dependencies.includes('next')) {
+      this.ctx.wizardData.detectedFramework = 'nextjs'
+    } else if (dependencies.includes('react-scripts')) {
+      this.ctx.wizardData.detectedFramework = 'cra'
+    } else if (dependencies.includes('nuxt')) {
+      this.ctx.wizardData.detectedFramework = 'nuxtjs'
+    } else if (dependencies.includes('@vue/cli-service')) {
+      this.ctx.wizardData.detectedFramework = 'vuecli'
+    }
+
+    if (this.ctx.wizardData.detectedFramework) {
+      return
+    }
+
+    if (dependencies.includes('react')) {
+      this.ctx.wizardData.detectedFramework = 'react'
+    } else if (dependencies.includes('vue')) {
+      this.ctx.wizardData.detectedFramework = 'vue'
+    }
+  }
+
+  private detectBundler (dependencies: string[]) {
+    const detectedFrameworkObject = FRONTEND_FRAMEWORKS.find((f) => f.type === this.ctx.wizardData.detectedFramework)
+
+    if (detectedFrameworkObject && detectedFrameworkObject.supportedBundlers.length === 1) {
+      this.ctx.wizardData.detectedBundler = detectedFrameworkObject.supportedBundlers[0] ?? null
+
+      return
+    }
+
+    if (dependencies.includes('webpack')) {
+      this.ctx.wizardData.detectedBundler = 'webpack'
+    }
+
+    if (dependencies.includes('vite')) {
+      this.ctx.wizardData.detectedBundler = 'vite'
+    }
+  }
+
+  private async detectLanguage () {
+    let hasTsConfig: boolean = false
+
+    try {
+      if (this.ctx.currentProject) {
+        hasTsConfig = !!await this.ctx.fs.stat(path.join(this.ctx.currentProject, 'tsconfig.json'))
```

Comment:
I feel like we have a "is it TS" check somewhere already from a PR Alejandro made?

Reply:
Yeah deep in the lifcycle manager. Should we make it a util?

Reply:
found it.